doc/docusaurus/docs/auction-smart-contract/on-chain-code.md

```diff
@@ -0,0 +1,157 @@
+---
+sidebar_position: 5
+---
+
+# On-chain Code: The Auction Validator
+
+:::caution
+This conceptual guide to an auction smart contract in Plutus introduces fundamentals for educational use.
+However, it is not optimized for security or efficiency and should not be deployed in production environments.
+Refer to resources like **[Cardano Plutus Script Vulnerability Guide](https://library.mlabs.city/common-plutus-security-vulnerabilities)** for best practices on developing secure smart contracts.
+:::
+
+# Auction Properties
+
+In this example, a seller wants to auction some asset she owns, represented as a non-fungible token (NFT) on Cardano.
+She would like to create and deploy an auction smart contract with the following properties:
+
+- there is a minimum bid amount
+- each bid must be higher than the previous highest bid (if any)
+- once a new bid is made, the previous highest bid (if exists) is immediately refunded
+- there is a deadline for placing bids; once the deadline has passed, new bids are no longer accepted, the asset can be transferred to the highest bidder (or to the seller if there are no bids), and the highest bid (if exists) can be transferred to the seller.
+
+# Plutus Tx Code
+
+Plutus Tx is a subset of Haskell, used to write on-chain code, also known as validators or scripts.
+A Plutus Tx program is compiled into Plutus Core, which is interpreted on-chain.
+The full Plutus Tx code for the auction smart contract can be found at [AuctionValidator.hs](https://github.com/IntersectMBO/plutus-tx-template/blob/main/src/AuctionValidator.hs).
+
+<!-- will need to update the link and file location for the new docs platform implementation -->
+
+## Data types
+
+First, let's define the following data types and instances for the validator:
+
+<LiteralInclude file="AuctionValidator.hs" language="haskell" title="Data types" start="-- BLOCK1" end="-- BLOCK2" />
+
+The purpose of `makeLift` and `makeIsDataSchemaIndexed` will be explained later.
```

Comment:
What about `HasBlueprintDefinition`. Also, I would put this before the code.

Reply:
`HasBlueprintDefinition` is generated by `makeIsDataSchemaIndexed`.